### ISSUE TITLE
Demo 1 feedback 1.0.0.4

### ISSUE THREAD

**Issue by mzorz** [Automattic/portkey-android#61]
Feedback from @megsfulton on the build made out of this tag https://github.com/Automattic/portkey-android/releases/tag/v1.0.0.4-demo1

Action items written by me after a first pass on understanding what was needed and how to do things.

**Camera**
- Does the camera library we're using have the ability to tap on the view to focus on an area? I keep finding myself trying to do it and miss having that functionality

Action item:
- [x] ~add a GH ticket to do later, as a quick search tells me this is doable but may be tricky, also not sure whether this is both available on Camera2 and CameraX as well (found examples for Camera2 only).~ issue opened https://github.com/Automattic/portkey-android/issues/62

**Video**
- I like the vibrate on video capture start but it's getting picked up in the recording audio so I think unless there's a way to prevent that we should remove it. 
- For the sound toggle, can we change the label to be "Sound on" / "Sound off" ?

Action items:
- [x] ~make the video recording start when the vibration ends~ 53f5a1a
- [x] ~change sound toggle label to "Sound on" / "Sound off"~ d1022d4

**Stickers**
- I LOVE that I can swipe down to dismiss the sticker list. Can it be accessed by a swipe up from the bottom of the screen? :D
- Can the sticker list start higher up on the screen until we implement a sheet view? 
- I have a hard time making the stickers smaller than their original size, not sure what could be done to make it easier to pinch to scale them down? 

Action items:
- [x] ~implement navigation bar visible first (look further down this list), then implement swipe from bottom gesture to make the bottom sheet appear (_follow up: This is more of a nice to have, fine with making an issue for it and revisiting it later._)~ implemented in a7a0936
- [x] Option 1. make the bottom sheet slide all the way up to the top of the screen when opening or ~Option 2. maybe then to make it clearer that it's a sheet we don't extend the bg to the top. I'll mock something up for us to chat about more~.  _EDIT Made the dialog appear expanded when opening ff45ab4_
- [x] ~this one sounds tricky, investigate pinch and see if it's possible to start detecting from a bit outside the original emoji size (i.e. surrounding it), but still make it scale down~ tracked in #66 to tackle later, labeled as "enhancement"

**Text editing**
- Increase width of the done button so there's at least 20 px padding on the left and right 
- Can the initial placement of a new text object retain the same text size as in the edit mode? (And then on subsequent edits it returns to the width / height / angle the user has set.) 
- Text color circles should be 24 x 24 _follow up as I intended to do it that way first but they ended up being too small to touch_: use a 40x40 touchable surface, and 16px between each circle as separation
<img width="266" alt="Screen Shot 2019-08-27 at 1 01 58 PM" src="https://user-images.githubusercontent.com/6597771/63854394-77369a80-c973-11e9-955a-a0ddabce9601.png">

Action items:
- [x] ~Increase width of the done button so there's at least 20 px padding on the left and right~ 40d8ee8
- [x] ~Increase the initial text size to keep the same one as in the Edit control~ 9a5a24f
- [x] ~make circle size and layout as per this: Text color circles should be 24 x 24, with a 40x40 touchable surface, and 16px between each circle as separation~ 2ec543e

**Saving**
- On the snackbar - update the label to "Saved to photos"  

Action item:
- [x] ~update snackbar label to "Saved to photos"~ c8d8856

**Deleting**
- Can the object that's being dragged to the delete button be in front of the delete button? (right now when you drag to the delete area the object is behind the button)

Action item:
- [x] ~make the delete button be added to the parentView from within PhotoEditor library so added views appear on top~ dd49e70 Also while at it, made the view that is about to be deleted semi-transparent in 0514e86, to not cover the Delete button and also to indicate it's fading away for good if you lift your finger there

**Status bar / os nav buttons**
- I'm wondering if we should display the software os buttons and status bar all the time. I find myself getting frustrated with the demo build because I can't always get the buttons to appear. And it will prevent them from hiding things / needing to have parts of the ui higher up (like the save button) 
_Follow up on this:  I want to keep the bar with the navigation buttons.[...] Let go with keeping the nav buttons on the screen all the time. That way we can use the swipe up to summon the sticker sheet [...] I'm ambivalent on the status bar, but if we do keep it hidden I like the idea of moving the top buttons down when it appears so it's not covering any screen ui_

Action item:
- [x] ~change flags in immersive UI setting to let navigation bar be always present, and status bar hidden - will follow up later to get another round of feedback once a new build is ready for Megs to check.~  done in 19fb394
Also related, added the following:
- [x] setup static image crop transform on Glide to properly show the captured image as per the editing screen background (f571651 and d7117f4)
- [x] adjusted the Save button to now have `16dp` bottom and right margin (anchored to bottom|right) (0d97144)
- [x] ~Still need to figure out how to move the top buttons down when the status bar appears so it's not covering any screen ui~ tracked here https://github.com/Automattic/portkey-android/issues/67

**Design changes (megs to-do list)** 
- Want to swap the flip icon for something that balances better with it's new placement opposite upload icon. Will provide something for that this week.  https://github.com/Automattic/portkey-android/issues/64
- Figure out a design / placement for the delete button https://github.com/Automattic/portkey-android/issues/63

Action items:
_None yet, will add here as design becomes available_

**Bugs?**
- Quit the app during editing mode on a photo. When I returned to the app, the camera is no longer "on" and I just see a black screen. (Have had this happen in all the builds so maybe a bug?) 
- When I quit the app with the keyboard up, the keyboard stays up over my app list view rather than disappearing with the rest of the app

Action items:
- [x] ~open issue for black screen problem &  tackle it~  opened #65 to tackle later as haven't been able to reproduce yet
- [x] ~open issue for keyboard up problem & tackle it~ fixed in 98f5c76 - this was super tricky to get working 😠-  need to test on various OS versions
